### DISCUSSION TITLE
Prefect Auth Proxy

### DISCUSSION THREAD

**Discussion by JoshuaSeidel** [PrefectHQ/prefect#5307] [Show and tell]
# Proxy Authorization Service for Prefect UI and Prefect CLI

[Prefect](https://prefect.io) is a great platform for building data flows/pipelines. It supports hybrid execution with execution engines running on-premises with
all command control and monitoring capabilities via the Prefect UI hosted on the cloud. This means metadata about flows, logs and metrics
will be stored in the cloud and can be accessed via the UI. When possible, we do recommend using the Prefect Cloud to manage your data flows.

However some organizations may prefer or required to run their flows on-premises and have the UI run on-premises as well. This is where the
Proxy Authorization Service comes in. Prefect currently do not bundle Authentication/Authorization options for API in the open source eco-system.
This functionality is only available with Prefect Cloud, along with many other features.

If your deployments have any constraints or security regulations to use Prefect Cloud, this Proxy service could be used as a workaround for
Prefect UI and CLI to enable authentication/authorization for API. All the requests to Prefect API will be proxied through this service.

[Softrams Prefect-Auth-Proxy](https://github.com/softrams/prefect-auth-proxy)
Main Contributor: @mkmurali

**Comment by gaby** [comment]
@JoshuaSeidel Thanks for sharing, any plans to do a new release? Last Docker image is from 2022
